azure_sdk_storage_core/src/connection_string.rs

```diff
@@ -0,0 +1,348 @@
+use azure_sdk_core::errors::AzureError;
+
+// Key names.
+pub const ACCOUNT_KEY_KEY_NAME: &str = "AccountKey";
+pub const ACCOUNT_NAME_KEY_NAME: &str = "AccountName";
+pub const SAS_KEY_NAME: &str = "SharedAccessSignature";
+pub const ENDPOINT_SUFFIX_KEY_NAME: &str = "EndpointSuffix";
+pub const DEFAULT_ENDPOINTS_PROTOCOL_KEY_NAME: &str = "DefaultEndpointsProtocol";
+pub const USE_DEVELOPMENT_STORAGE_KEY_NAME: &str = "UseDevelopmentStorage";
+pub const DEVELOPMENT_STORAGE_PROXY_URI_KEY_NAME: &str = "DevelopmentStorageProxyUri";
+pub const BLOB_ENDPOINT_KEY_NAME: &str = "BlobEndpoint";
+pub const BLOB_SECONDARY_ENDPOINT_KEY_NAME: &str = "BlobSecondaryEndpoint";
+pub const TABLE_ENDPOINT_KEY_NAME: &str = "TableEndpoint";
+pub const TABLE_SECONDARY_ENDPOINT_KEY_NAME: &str = "TableSecondaryEndpoint";
+pub const QUEUE_ENDPOINT_KEY_NAME: &str = "QueueEndpoint";
+pub const QUEUE_SECONDARY_ENDPOINT_KEY_NAME: &str = "QueueSecondaryEndpoint";
+pub const FILE_ENDPOINT_KEY_NAME: &str = "FileEndpoint";
+pub const FILE_SECONDARY_ENDPOINT_KEY_NAME: &str = "FileSecondaryEndpoint";
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum ConnectionStringError {
+        MissingValue { key: String } {
+            display("Missing value for key '{}'", key)
+        }
+        UnexpectedKey { key: String } {
+            display("Unexpected for key '{}'", key)
+        }
+        ParsingError { msg: String } {
+            display("{}", msg)
+        }
+    }
+}
+
+impl From<ConnectionStringError> for AzureError {
+    fn from(err: ConnectionStringError) -> Self {
+        AzureError::GenericErrorWithText(err.to_string())
+    }
+}
```

Comment:
@MindFlavor Haven't seen this done previously in the codebase, if there's a better way to convert between the errors let me know :)
Didn't want to create a whole new type in `AzureError`.

Reply:
Sounds good! As of now the `AzureError` story needs some rethinking so I would not fret too much about it. Maybe we can encapsulate your error type in AzureError enum in the future (this probably means moving it to the `core ` crate). [https://github.com/MindFlavor/AzureSDKForRust/blob/69337ec4f6c62fbb87de158471c7b5e3b24b2eba/azure_sdk_core/src/errors.rs#L144-L285](https://github.com/MindFlavor/AzureSDKForRust/blob/69337ec4f6c62fbb87de158471c7b5e3b24b2eba/azure_sdk_core/src/errors.rs#L144-L285)